Update README.md documentation

miguelsorianod opened the pull request:
I added in the README.md how to deploy an empty 3scale-operator, and refer to the user-guide on more details on what it does and how to deploy some example custom resources for it.

mikz commented:
Why into README.md ? Bloated readmes are quite off putting. Each part of the readme can be independent document and the README acts as just as a table of contents.

miguelsorianod commented:
I agree. For the moment I rolled back the user-guide content, kept the contributing section and added a reference to the user-guide from the quickstart.

miguelsorianod commented:
How about now?

mikz commented on README.md:
```diff
@@ -1,25 +1,77 @@
-$ cd 3scale-operator
-$ git checkout master
-$ make vendor
+mkdir -p $GOPATH/src/github.com/3scale
```
Can't you just `go get -u && cd`?

miguelsorianod replied:
It seems that the structure of the project is not completely go-gettable (the project structure is generated by the operator-sdk new command):

```
msoriano@localhost:~$ go get -u github.com/3scale/3scale-operator
package github.com/3scale/3scale-operator: no Go files in /home/msoriano/exampleworkspace/src/github.com/3scale/3scale-operator
```

I've seen other projects doing it the way I wrote too (operator-sdk for example), maybe for the same reason.

mikz commented on README.md:
````diff
@@ -1,25 +1,77 @@
+Download the 3scale-operator project and install the needed dependencies:
+
+```sh
+mkdir -p $GOPATH/src/github.com/3scale
````
This is the same as above. Should not be needed to repeat.

miguelsorianod replied:
There was one additional step. I modified the content to refer to the other section from here and just add the additional step

miguelsorianod commented:
I've updated the PR again.
I've now added in the README.md how to deploy an empty 3scale-operator, and refer to the user-guide on more details on what it does and how to deploy some example custom resources for it.

The idea is to add the remaining Capabilities part in the user-guide @jmprusi  in another PR, and also the corresponding Capabilities-related spec file/s and reference them from the README.md.
